authFunctions/login.go

```diff
@@ -0,0 +1,111 @@
+package authFunctions
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/dgrijalva/jwt-go"
+	_ "github.com/go-sql-driver/mysql"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func createToken(user User) (string, error) {
+	var err error
+
+	// 鍵となる文字列(多分なんでもいい)
+	secret := "auth-using-go"
+
+	// Token を作成
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+		"email": user.Email,
+		"iss":   "FunCloudAPI", // JWT の発行者が入る(文字列は任意)
+	})
+
+	//Dumpを吐く
+	spew.Dump(token)
+
+	tokenString, err := token.SignedString([]byte(secret))
+
+	fmt.Println("-----------------------------")
+	fmt.Println("tokenString:", tokenString)
+
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
先にエラーハンドリングしたほうが良いと思う。エラー起きてるtokenStringを出力する意味が無いからです。
```suggestion
	tokenString, err := token.SignedString([]byte(secret))

	if err != nil {
		log.Fatal(err)
	}

	fmt.Println("-----------------------------")
	fmt.Println("tokenString:", tokenString)
```

Reply:
あと、log.Fatalってプログラムを終了してしまうので、ここではlog.Fatalしないほうが良いです。[参考](https://golang.org/src/log/log.go#L332)os.Exit(1)はプログラムを終了する関数です。


つまりどうすればいいかというと以下です。
```
	if err != nil {
		log.Println(err)
                 return "", err
	}
```

Reply:
` return "", err`は，なにを表しているのでしょうか．．．？
単に関数の終了`return`だけの場合との違いは何でしょうか？